### PR TITLE
Canada: Fixes PHP notices, handles form unknown errors.

### PR DESCRIPTION
**Pull request by sergiitk**
#### What's this PR do?
- Fixes PHP notices on user registration:
  screen shot 2014-11-19 at 11 36 57 pm
- Handles form unknown errors:
  image
#### Testing

Tests are consistent:
image

### CODE REVIEW

File: lib/modules/dosomething/dosomething_canada/includes/dosomething_canada_register_controller.inc [hunk 1]
```diff
@@ -30,36 +30,41 @@ class DosomethingCanadaRegisterController implements ExternalAuthRegisterControl
   }
 
   public function signup() {
-    $user = new stdClass();
-    $user->email    = $this->account_values['mail'];
-    $user->password = $this->account_values['pass'];
+    $local_user = new stdClass();
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_canada/includes/dosomething_canada_register_controller.inc:33]: Just checking, this name change is just to make the code more readable?  Where are the big changes here that fix things?

File: lib/modules/dosomething/dosomething_canada/includes/dosomething_canada_register_controller.inc [hunk 2]
```diff
@@ -30,36 +30,41 @@ class DosomethingCanadaRegisterController implements ExternalAuthRegisterControl
...
     }
 
     if (!empty($this->account_values['field_last_name']['und'][0])) {
-      $user->lastName = $this->account_values['field_last_name']['und'][0]['value'];
+      $local_user->lastName = $this->account_values['field_last_name']['und'][0]['value'];
     }
 
     $result = FALSE;
     try {
-      $result = $this->sso->propagateLocalUser($this->email, $this->password, $user);
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_canada/includes/dosomething_canada_register_controller.inc:55]: It's here: there was no `$this->email` and `$this->password`, my fault.

File: lib/modules/dosomething/dosomething_canada/includes/dosomething_canada_register_controller.inc (hunk 1 above)
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_canada/includes/dosomething_canada_register_controller.inc:33]: See above.

**Comment by aaronschachter** [comment]
:+1:

File: lib/modules/dosomething/dosomething_canada/includes/dosomething_canada_register_controller.inc [hunk 3]
```diff
@@ -30,36 +30,41 @@ class DosomethingCanadaRegisterController implements ExternalAuthRegisterControl
...
     } catch (Exception $e) {
       return FALSE;
     }
     return $result;
   }
 
-  public function processSignupErrors(Array $form) {}
+  public function processSignupErrors(Array $form) {
+    // SSO returned unexpected errors with no much to local fields.
+    if (!form_get_errors()) {
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_canada/includes/dosomething_canada_register_controller.inc:64]: And here. Without that all request errors are will be just ignored and user actually will be created.

**Pull request merged**